survey_certification_sending/models/survey_user_input.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2025 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, fields, models
+
+
+class SurveyUserInput(models.Model):
+    _inherit = "survey.user_input"
+
+    certification_sent = fields.Boolean(
+        default=False, help="It indicates if the certification has been sent."
+    )
+
+    def _mark_done(self):
+        for user_input in self:
+            if (
+                not user_input.survey_id.skip_certification_email
+                and user_input.survey_id.certification
+                and user_input.scoring_success
+            ):
+                if (
+                    user_input.survey_id.certification_mail_template_id
+                    and not user_input.test_entry
+                ):
+                    user_input.certification_sent = True
```

Comment:
This is to be flagged just when the certification mail is sent

Reply:
It has been discussed in an internal conversation to reach the agreement to indicate with a comment this operation at this point in the logic.